src/main/scala/vectorpipe/VectorPipe.scala

```diff
@@ -141,7 +142,7 @@ object VectorPipe {
       val simplify = udf { g: jts.Geometry => pipeline.simplify(g, level.layout) }
       val reduced = pipeline
         .reduce(working, level, keyColumn)
-        .localCheckpoint()
+        .persist(StorageLevel.MEMORY_AND_DISK_SER)
```

Comment:
Healthy skepticism.  I've commented this out in order to complete a global-scale ingest (that never finished in either case).  Seems like local checkpointing will have a similar memory pressure, though.  I can't seem to find any documentation that suggests that checkpoints are less memory consumptive?  I have no idea what the implications of a truncated lineage are, though.  It's not well described anywhere I've looked.  It may be best to use a flag to indicate which caching strategy to use, if any, depending on the input job size.  All I know is that global jobs are memory hogs.  I used 32 r5.2xlarges, and still needed to add on a half terabyte EBS volume to each to get the job done (though it still failed for non-memory-related reasons).

Reply:
Follow up: it appears that checkpointing in spark demands a caching (to memory) operation or the checkpoint will trigger a recomputation.  See the first note [here](https://jaceklaskowski.gitbooks.io/mastering-apache-spark/spark-rdd-checkpointing.html).

I've since run some processes based on `OSM.toGeometry` which were structured similar to `VectorPipe.apply` which triggered recomputations without some kind of persistence.  As suspected, caching to memory was a fool's errand.  A fool's errand!  And I was reminded that I indeed do have a funny hat.  So, the happy intermediate ground is to persist to `DISK_ONLY`.  It is also possible to cache to memory the result of `OSM.toGeometry` prior to entering `VectorPipe.apply` without causing big trouble.  It's a good tradeoff of performance to memory pressure.  

The long and short is that one may need more disk/memory on the executors, but it's easy to add EBS volumes, and it also seems as if a properly sized cluster of r-series nodes will do the job just fine without blowing up or needing to pay for excess compute cycles just to increase the available cluster memory.  Whether that scales to global ingests is anyone's guess, though.